main/document/edit_document.php

```diff
@@ -412,28 +431,59 @@
         }
     }
 
-    if (!$group_document && !DocumentManager::is_my_shared_folder(api_get_user_id(), $currentDirPath, $sessionId)) {
-        $form->addLabel(get_lang('UpdatedOn'), Display::dateToStringAgoAndLongDate($last_edit_date));
-    }
 
-    if (!empty($document_info['insert_user_id'])) {
-        $insertByUserInfo = api_get_user_info($document_info['insert_user_id']);
-        if (!empty($insertByUserInfo)) {
-            $form->addLabel(get_lang('Author'), $insertByUserInfo['complete_name_with_message_link']);
+    if ($file_type != 'link') {
+        if (!$group_document && !DocumentManager::is_my_shared_folder(api_get_user_id(), $currentDirPath, $sessionId)) {
+            $form->addLabel(get_lang('UpdatedOn'), Display::dateToStringAgoAndLongDate($last_edit_date));
+        }
+
+        if (!empty($document_info['insert_user_id'])) {
+            $insertByUserInfo = api_get_user_info($document_info['insert_user_id']);
+            if (!empty($insertByUserInfo)) {
+                $form->addLabel(get_lang('Author'), $insertByUserInfo['complete_name_with_message_link']);
+            }
         }
     }
 
-    $form->addElement('textarea', 'comment', get_lang('Comment'), ['cols-size' => [2, 10, 0]]);
+    if ($file_type == 'link') {
+        // URLs in whitelist
+        $urlWL = URLUtils::getFileHostingsWL();
+        sort($urlWL);
+        $urlWLRegEx = '/(\/\/|\.)('.implode('|', $urlWL).')/i'; //Matches any of the whitelisted urls preceded by // or .
+        $urlWLText = "\n\t* ".implode("\n\t* ", $urlWL);
+        $urlWLHTML = "<ul><li>".implode("</li><li>", $urlWL)."</li></ul>";
+        $form->addText('comment', get_lang('Url'));
+        $form->addElement('static', 'info', '', '<span class="text-primary" data-toggle="tooltip" title="'.$urlWLHTML.'">'.get_lang('ValidDomainsList').' <span class="glyphicon glyphicon-question-sign"></span></span>');
+    } else {
+        $form->addElement('textarea', 'comment', get_lang('Comment'), ['cols-size' => [2, 10, 0]]);
+    }
 
-    if ($owner_id == api_get_user_id() || api_is_platform_admin()) {
-        $checked = & $form->addElement('checkbox', 'readonly', null, get_lang('ReadOnly'));
-        if ($readonly == 1) {
-            $checked->setChecked(true);
+    if ($file_type != 'link') {
+        if ($owner_id == api_get_user_id() || api_is_platform_admin()) {
+            $checked = & $form->addElement('checkbox', 'readonly', null, get_lang('ReadOnly'));
+            if ($readonly == 1) {
+                $checked->setChecked(true);
+            }
         }
     }
 
+    if ($file_type == 'link') {
+        $form->addRule('title', get_lang('PleaseEnterCloudLinkName'), 'required', null, 'client');
+        $form->addRule('title', get_lang('PleaseEnterCloudLinkName'), 'required', null, 'server');
+        $form->addRule('comment', get_lang('langGiveURL'), 'required', null, 'client');
+        $form->addRule('comment', get_lang('langGiveURL'), 'required', null, 'server');
```

Comment:
don't use "lang" prefixes. This is an old standard that we are not using anymore. Maybe just "PleaseGiveUrl"?